### ISSUE TITLE
Issue with boundary[] in hide middleware

### ISSUE THREAD

**Issue by arunks5** [floating-ui/floating-ui#3108]
When we place a modal inside a drawer and then place a panel inside the drawer, the floating UI position becomes incorrect.

All use popover attribute and uses top-layer.

To overcome this , we used the solution suggested in https://github.com/floating-ui/floating-ui/issues/2858

As per the suggestion we used empty boundary (boundary:[]) for both shift and hide middlewares. It is working fine , but we are facing an issue in hide middleware.

We are hiding the floating element using
        contentElement.style.display = middlewareData.hide?.referenceHidden ? 'none' : 'block';
in compute Position
        
But, since boundary is empty, `middlewareData.hide?.referenceHidden` returns true only when the trigger element goes out of viewport. But, we need to hide the floating element when trigger element goes out the clipping ancestor.

When we remove the empty boundary in hide middleware, floating element doesnot display on larger screens, because
`middlewareData.hide?.referenceHidden` always returns true as given in https://github.com/floating-ui/floating-ui/issues/2858. We are currently using "@floating-ui/dom": "^1.6.11",

**To Reproduce**

I have created a repo [here ](https://stackblitz.com/edit/vitejs-vite-bmk9sy?file=src%2Fdrawer.ts,src%2Fgreet.ts,package.json,src%2Ffloating.ts,src%2Fpicker.ts)

1. Select 'left' radio button which passes, left as placement in computePosition
2. Click on show drawer
3. Click on the input-field
4. The floating element is displayed 
5. Scroll till the input goes out of view. 

Expected : 
As soon as the input field goes out of view, floating element should be hidden with this condition
        contentElement.style.display = middlewareData.hide?.referenceHidden ? 'none' : 'block';

Actual :
   Floating element is not hidden until input goes out of the entire viewport, not just out of  the clipping ancestor

- OS: [Windows]
- Browser [Chrome]
- Version [130]

**Comment by atomiks** [comment]
So the issue that was closed didn't fully fix the issue then? It still incorrectly calculates `referenceHidden` in this layout.

Technically there's a workaround to manually pass the scrolling element to `hide`'s `boundary` option.

**Comment by arunks5** [comment]
> So the issue that was closed didn't fully fix the issue then? It still incorrectly calculates `referenceHidden` in this layout.
Yes @atomiks , might be due to the css `transform: translate(-50%,-50%)`.
> Technically there's a workaround to manually pass the scrolling element to `hide`'s `boundary` option.
Unfortunately, the scrolling element is dynamic and can be anything. So, we couldn't pass the scrolling element to hide's boundary

**Comment by atomiks** [comment]
This function isn't documented, but there's an exported util called `getOverflowAncestors`.

In this case, the trigger's first one is the scrolling parent.

```js
import {getOverflowAncestors} from '@floating-ui/dom';
```

This works:
```js
const [scrollParent] = getOverflowAncestors(trigger);

hide({ boundary: scrollParent })
```

Won't work reliably in every case and may trigger false positives, but it's a workaround for this one at least.
